addons/website_sale/controllers/main.py

```diff
@@ -425,7 +425,12 @@ def shop(self, page=0, category=None, search='', min_price=0.0, max_price=0.0, p
         if filter_by_tags_enabled and search_product:
             all_tags = ProductTag.search(
                 expression.AND([
-                    [('product_ids.is_published', '=', True), ('visible_on_ecommerce', '=', True)],
+                    [
+                        ('visible_on_ecommerce', '=', True),
+                        '|',
+                        ('product_template_ids.is_published', '=', True),
+                        ('product_ids.is_published', '=', True),
```

Comment:
```suggestion
                        ('product_product_ids.is_published', '=', True),
```
I believe there is no need to search on `product_ids` anymore, it would be converted to `['|', ('product_template_ids.product_variant_ids', operator, operand), ('product_product_ids', operator, operand)]` (through the search method of the field).
Part of this converted domain would be useless because the `is_published` field is defined on the template anyway 🤔